clusterloader2/pkg/measurement/util/runtimeobjects/runtimeobjects.go

```diff
@@ -0,0 +1,218 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package runtimeobjects
+
+import (
+	"fmt"
+	"reflect"
+	"strconv"
+
+	batch "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+	extensions "k8s.io/api/extensions/v1beta1"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+	clientset "k8s.io/client-go/kubernetes"
+)
+
+// GetObjectForKind returns object instance of given kind.
+func GetObjectForKind(kind string) (runtime.Object, error) {
+	switch kind {
+	case "ReplicationController":
+		return &corev1.ReplicationController{}, nil
+	case "ReplicaSet":
+		return &extensions.ReplicaSet{}, nil
+	case "Deployment":
+		return &extensions.Deployment{}, nil
+	case "DaemonSet":
+		return &extensions.DaemonSet{}, nil
+	case "Job":
+		return &batch.Job{}, nil
+	default:
+		return nil, fmt.Errorf("unsupported kind when getting runtime object: %v", kind)
+	}
+}
+
+// GetRuntimeObjectsForKind returns objects of given kind that satisfy given namespace, labelSelector and fieldSelector.
+func GetRuntimeObjectsForKind(c clientset.Interface, kind, namespace, labelSelector, fieldSelector string) ([]runtime.Object, error) {
+	listOpts := metav1.ListOptions{
+		LabelSelector: labelSelector,
+		FieldSelector: fieldSelector,
+	}
+	runtimeObjectsList := make([]runtime.Object, 0)
```

Comment:
This may mean that you can do 2x more allocations than really needed.

Instead, you should allocated array when you know how many elements you have and in default case return nil.

Reply:
Done.